[Rubesh] iP

sp4ce-cowboy opened the pull request:
# ChatterBox

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

ChatterBox frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER FAST_ to use
- All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/490).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE!**

Features:

- [x]  Managing tasks
- [x]  Managing deadlines (coming soon)
- [x]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

sp4ce-cowboy committed:
Implement an initial skeletal version of the ChatBot. Duke Level-0, first commit

sp4ce-cowboy committed:
Delete Duke.java

sp4ce-cowboy committed:
Improve ChatBot to echo every input

sp4ce-cowboy committed:
Add ability to read, store, and retrieve tasks upon request

sp4ce-cowboy committed:
Add ability to mark and unmark tasks

sp4ce-cowboy committed:
Add ability to add todos, deadlines and event, in addition to tasks

sp4ce-cowboy committed:
Use the input/output redirection technique to semi-automate the testing of Duke

sp4ce-cowboy committed:
Teach the chatbot to deal with errors such as incorrect inputs entered by the user

sp4ce-cowboy committed:
Add support for deleting tasks from the list

sp4ce-cowboy committed:
Fix formatting errors.

sp4ce-cowboy committed:
Add support for file storage

sp4ce-cowboy committed:
Remove empty FileReader file.

sp4ce-cowboy committed:
Fixed support for file storage

sp4ce-cowboy committed:
Update .gitignore file to ignore .txt files

sp4ce-cowboy committed:
Add LocalDate functionality to Deadline Objects

sp4ce-cowboy committed:
Update .gitignore

sp4ce-cowboy committed:
Clean up code in Storage.java

sp4ce-cowboy committed:
Add UI class to deal with User Interface

sp4ce-cowboy committed:
Add more UI functionality and Storage class

sp4ce-cowboy committed:
Add more TaskList functionality and simplify main Class

sp4ce-cowboy committed:
Add more Parser Functionality

sp4ce-cowboy committed:
Add final OOP Functionalities

sp4ce-cowboy committed:
Package all classes into a single duke package.

sp4ce-cowboy committed:
Package more file into duke folder

sp4ce-cowboy committed:
Merge branch 'add-gradle-support'

sp4ce-cowboy committed:
Edit main file and add gradle support

sp4ce-cowboy committed:
Configure Gradle to run the ChatBot class

sp4ce-cowboy committed:
Add JUnit tests for 3 classes

sp4ce-cowboy committed:
Modify build.gradle file to support making of Jar file

sp4ce-cowboy committed:
Add JavaDoc comments to all classes

sp4ce-cowboy committed:
Format all code to conform to coding standards.

sp4ce-cowboy committed:
Add some changes to TaskList to accodomate Find functionality.

sp4ce-cowboy committed:
Merge branch 'branch-A-CodingStandard'

sp4ce-cowboy committed:
Add completed Find task functionality

sp4ce-cowboy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

sp4ce-cowboy committed:
Merge more conflicts

sp4ce-cowboy committed:
Clean up code

sp4ce-cowboy committed:
Edit some code for creation of Level-8 branch

WangCheng0116 commented on src/main/java/duke/Storage.java:
Would it be better if you could separate Storage into another file or package?

WangCheng0116 commented on src/main/java/duke/ChatterBox.java:
```diff
@@ -0,0 +1,124 @@
+            String command = p.command();
+            String fullLine = p.fullLine;
+
+            if (command.equals("bye")) {
```
Would it be better if we separate all these if-else statements into Parser?

WangCheng0116 commented on src/main/java/duke/TaskList.java:
I like how nice and clean this class is. Well done! 👍

WangCheng0116 commented on src/test/java/duke/UiTest.java:
```diff
@@ -0,0 +1,44 @@
+        System.setOut(originalOut);
+    }
+
+    @Test
```
I think including more thorough testings would be great.

WangCheng0116 commented on src/main/java/duke/Task.java:
LGTM

WangCheng0116 commented on src/main/java/duke/ToDo.java:
LGTM

WangCheng0116 reviewed (commented):
Overall, your code is very easy to understand and it follows the coding standard. The main logic is clearly organized. But for a few components, I think it would be a better idea to separate them into another package, such as UI or Storage.

freshcabbage123 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,20 @@
+package duke;
+
+import java.util.*;
+import java.lang.Exception;
```
Is it possible for the code to run even without importing java.lang.Exception? I believe it should be possible. To nitpick, if java.lang.Exception is to be imported, it should be done so before java.util.* 👍

freshcabbage123 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,80 @@
+ */
+public class Task {
+    boolean done;
+    String name;
```
Should access modifiers be given to variables 'done' and 'name'? I believe they should be private and given appropriate accessor methods if necessary.

freshcabbage123 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,65 @@
+ */
+public class Deadline extends Task {
+
+    LocalDate deadline;
```
Similar issue as earlier, I believe access modifiers should be added?

freshcabbage123 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+
+import java.io.File;
+import java.io.FileWriter;
+
```
Is this line break meant to separate the imports? I believe it won't be necessary as the import on line 4 and line 6 are part of the same group 👍

freshcabbage123 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+        String longName = fullLine.split("/by ")[0];
+        String date = fullLine.split("/by ")[1];
+        String taskName = longName.split("deadline ")[1];
+        String deadlineName = taskName +
```
Minor nitpick but i believe the '+' operator should be on a new line 👍

freshcabbage123 reviewed (commented):
Overall, looks good, just minor nitpicks for the sake of the assignment!

sp4ce-cowboy committed:
Modify gradle build file and java files to prepare for JavaFX implementation

sp4ce-cowboy committed:
Add GUI functionality to ChatterBox

sp4ce-cowboy committed:
Modify JUnit tests to accomodate new GUI functionality

sp4ce-cowboy committed:
Clean up some code in Main class

sp4ce-cowboy committed:
Modify gradle build file

The gradle build file contains references to the previous default duke
project.

Modifying this build file to convert all these references and the name
of the jar file to chatterbox will unify the working environment.

sp4ce-cowboy committed:
Add assertions functionality

Until now, it has been assumed that all interacting methods correctly
provide the data required by the calling method.

To address any possible programmer errors and oversight, assertions have
been inserted at appropriate junctures in the code for all relevant
classes.

This will make the application run more safely and errors will be
explicitly highlighted instead of being suppressed when the app is run.

sp4ce-cowboy committed:
Clean up code

The codebase up until this point did not entire fit in with the required
Coding Quality requirements.

All Magic Numbers are replaced with final class constants, and other
minor code quality issues are fixed.

sp4ce-cowboy committed:
Merge pull request #1 from sp4ce-cowboy/branch-A-Assertions

Merge commit to add assertions functionality

sp4ce-cowboy committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions functionality

sp4ce-cowboy committed:
Merge PRs

Clean up code, merge pull request #2 from sp4ce-cowboy/branch-A-CodeQuality

sp4ce-cowboy committed:
Initialize ability to Update items

sp4ce-cowboy committed:
Add support for mass deletion

sp4ce-cowboy committed:
Improve text formatting

sp4ce-cowboy committed:
Improve exception handling and overall robustness

sp4ce-cowboy committed:
Update images for User and Duke

sp4ce-cowboy committed:
Finalize chatterbox for submission

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Improve code syntax across Tasks classes

Until now, each Task child class uses magic literals in the
enumeration of its Task type.

These literals have been replaced by static final Strings in each
class, allowing for better adherence to the coding standards.

sp4ce-cowboy committed:
Update gradle build file

The previous iteration did not work on non-macOS environments.

This was caused by errors with the wrong mainClass being loaded.

This has been fixed by changing the mainClass to Launcher instead.

sp4ce-cowboy committed:
Extract parseText method into Parser class

sp4ce-cowboy committed:
Modify app to show exit screen before exiting

sp4ce-cowboy committed:
Add checkstyle to gradle build file

sp4ce-cowboy committed:
Update code to conform with CheckStyle

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Add help functionality

sp4ce-cowboy committed:
Merge branch 'master' of https://github.com/sp4ce-cowboy/ip

* 'master' of https://github.com/sp4ce-cowboy/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md

sp4ce-cowboy committed:
Extract all parser methods and account for fringe Exceptions

sp4ce-cowboy committed:
Extract out more commands into the CommandParser class

sp4ce-cowboy committed:
Add JUnit tests for Parser class

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Add Help Command instructions

The previous iteration contained the functionality for the help
command but did not mention it specifically.

I change the Ui Class to include the help function explicitly.

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md

sp4ce-cowboy committed:
Update README.md
